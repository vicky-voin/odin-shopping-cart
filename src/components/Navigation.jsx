import { ShoppingCart } from "lucide-react";
import styles from "./Navigation.module.css";
import { NavLink } from "react-router";

export default function Navigation({ cartData }) {
  const totalCount =
    cartData == null
      ? 0
      : cartData.reduce((acc, item) => acc + item.quantity, 0);

  function getNavLinkClass({ isActive }) {
    return isActive ? `${styles.navButton} ${styles.active}` : styles.navButton;
  }

  return (
    <nav className={styles.root}>
      <NavLink className={getNavLinkClass} data-testid="homeNavLink" to="/">
        Home
      </NavLink>
      <NavLink className={getNavLinkClass} data-testid="shopNavLink" to="/shop">
        Shop
      </NavLink>
      <NavLink className={getNavLinkClass} data-testid="cartNavLink" to="/cart">
        <ShoppingCart></ShoppingCart>
        <span data-testid="cartTotalCount">{totalCount}</span>
      </NavLink>
    </nav>
  );
}
