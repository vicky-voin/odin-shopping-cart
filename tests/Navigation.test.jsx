import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Navigation from "../src/components/Navigation";
import { MemoryRouter } from "react-router";

describe("Navigation component tests", () => {
  it("renders navigation links", () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation cartData={[]}></Navigation>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
