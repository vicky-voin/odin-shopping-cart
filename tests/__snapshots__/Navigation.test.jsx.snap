// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation component tests > renders navigation links 1`] = `
<div>
  <nav
    class="_root_b30331"
  >
    <a
      aria-current="page"
      class="_navButton_b30331 _active_b30331"
      data-discover="true"
      data-testid="homeNavLink"
      href="/"
    >
      Home
    </a>
    <a
      class="_navButton_b30331"
      data-discover="true"
      data-testid="shopNavLink"
      href="/shop"
    >
      Shop
    </a>
    <a
      aria-label="Cart, total count 0"
      class="_navButton_b30331"
      data-discover="true"
      data-testid="cartNavLink"
      href="/cart"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-shopping-cart"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="21"
          r="1"
        />
        <circle
          cx="19"
          cy="21"
          r="1"
        />
        <path
          d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
        />
      </svg>
      <span
        data-testid="cartTotalCount"
      >
        0
      </span>
    </a>
  </nav>
</div>
`;
